executor/analyze.go

```diff
@@ -897,8 +904,34 @@ func (e *AnalyzeFastExec) handleSampTasks(bo *tikv.Backoffer, workID int, err *e
 }
 
 func (e *AnalyzeFastExec) buildHist(ID int64, collector *statistics.SampleCollector, tp *types.FieldType) (*statistics.Histogram, error) {
-	// TODO: build histogram and cmsketch here for one collector.
-	return nil, nil
+	// build collector properties.
+	collector.Samples = collector.Samples[:e.sampCursor]
+	err := collector.CalcTotalSize()
+	if err != nil {
+		return nil, err
+	}
+	collector.Count = int64(e.sampCursor)
+	data := make([][]byte, 0, len(collector.Samples))
+	for _, sample := range collector.Samples {
+		if sample.Value.IsNull() {
+			collector.NullCount++
```

Comment:
For multi-column index, its `NullCount` should be kept as 0 and not be increased.

Reply:
If the sample is established by multi-columns index. The `IsNull()` check is always false?

Reply:
Oh yes you are right.